backend/eduschola_project/edusch_app/models.py

```diff
@@ -54,49 +55,75 @@ def __str__(self):
 
 class Grade(models.Model):
     grade_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    name = models.CharField(max_length=255)
-    school = models.ForeignKey(School, on_delete=models.CASCADE)  # Many-to-One relationship with School model
+    staff = models.ForeignKey('Staff', on_delete=models.SET_NULL, null=True)
+    student = models.ForeignKey('Student', on_delete=models.SET_NULL, null=True)
+    course = models.ForeignKey(Course, on_delete=models.SET_NULL, null=True)
+    term = models.CharField(max_length=255, default='')
+    session = models.CharField(max_length=255, default='')
+    grade = models.FloatField(default=0)
 
     def __str__(self):
-        return self.name
+        return f"Grade {self.grade_id} - Staff: {self.staff}, Student: {self.student}, Course: {self.course}"
 
 
 class Student(models.Model):
     student_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    user = models.OneToOneField('User', on_delete=models.CASCADE, related_name='student_user')
+    user = models.OneToOneField('User', on_delete=models.CASCADE, null=True, related_name='student_user')
     date_of_birth = models.DateField()
     phone_number = models.TextField()
     address = models.TextField()
-    parent = models.ForeignKey('Parent', on_delete=models.CASCADE)
-    school = models.ForeignKey(School, on_delete=models.CASCADE)  # Many-to-One relationship with School model
-    grade = models.ForeignKey(Grade, on_delete=models.CASCADE)  # Many-to-One relationship with Class model
+    parent = models.ForeignKey('Parent', on_delete=models.SET_NULL, null=True, blank=True)
+    school = models.ForeignKey(School, on_delete=models.SET_NULL, null=True)  # Many-to-One relationship with School model    
+
+    def delete(self, *args, **kwargs):
+        # Delete associated User instance
+        if self.user:
+            self.user.delete()
+
+        super().delete(*args, **kwargs)
 
     def __str__(self):
         return self.user.username
 
 
-class Instructor(models.Model):
-    instructor_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    user = models.OneToOneField('User', on_delete=models.CASCADE, related_name='instructor_user')
+class Staff(models.Model):
+    staff_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    user = models.OneToOneField('User', on_delete=models.CASCADE, null=True, related_name='staff_user')
+
     qualification = models.CharField(max_length=255)
```

Comment:
Dont think qualification is a necessary information to store. I think something better to store would be their level. Maybe if they were teaching or non teaching staff. or maybe assistant professors or professors